StortSpelProjekt/Engine/src/Particles/ParticleSystem.cpp

```diff
@@ -15,17 +19,26 @@ ParticleSystem& ParticleSystem::GetInstance()
 
 ParticleSystem::ParticleSystem()
 {
-	effect = new ParticleEffect();
+	Renderer& renderer = Renderer::GetInstance();
 }
 
 ParticleSystem::~ParticleSystem()
 {
-	delete effect;
 }
 
 void ParticleSystem::Update(double dt)
 {
-	effect->Update(dt);
+	if (effect != nullptr)
+	{
+		//for(ParticleEffect& particleEffect : )
+		effect->Update(dt);
+		effect->updateResourceData();
+	}
+}
+
+void ParticleSystem::SetParticleEffect(ParticleEffect* effect)
+{
+	this->effect = effect;
```

Comment:
With correct code standard, 'this' is not necessary

Reply:
fixed